### ISSUE TITLE
When reloading the watchlist is gone

### ISSUE THREAD

**Issue by K3NOXOFFICIAL** [Ashanime/Ashanime-Web-App#7]
When I reload the website my watchlist is gone.
Any fix?

**Comment by ahmedalashhab** [comment]
Hey there,
The reason for this and your other issue regarding firebase config is that you need to create a firebase account and your own firebase config file. You can find how to do this in Ashanime's Read Me. Please do follow the guide, if that does not work, I can help you further. Firebase acts as a back end on the cloud and it stores all your user data, including your watchlist to the firebase backend.

**Comment by K3NOXOFFICIAL** [comment]
Hey, I created a firebase app changed the config to my own details and it still doesn't work

**Comment by ahmedalashhab** [comment]
I'm sorry for the delays on my end. I'll make sure to solve your issues ASAP. It's been a hectic week for me. Hopefully sometime this week.

**Comment by K3NOXOFFICIAL** [comment]
Can you also if you have time answer my email by any chance and update the discord invite link for better support?
Thanks a lot btw.

**Comment by ahmedalashhab** [comment]
Is this working well now since Firebase is working with you?

**Comment by K3NOXOFFICIAL** [comment]
Nope tried it on the ashanime website, my own deployed one and it doesn't work anywhere.
Phone, pc and laptop so it's no cookie or browser related problem.

**Comment by K3NOXOFFICIAL** [comment]
https://user-images.githubusercontent.com/46091052/204399771-b053482b-6f3f-4a40-b088-86952e5dccf0.mp4

**Comment by K3NOXOFFICIAL** [comment]
@ahmedalashhab  you can see it just disappears after reloading the website.

https://user-images.githubusercontent.com/46091052/204400426-65cf40eb-4c78-476d-bf4e-dacf3d9cbf91.mp4

**Comment by K3NOXOFFICIAL** [comment]
@ahmedalashhab Nope tried it on the ashanime website, my own deployed one and it doesn't work anywhere.
Phone, pc and laptop so it's no cookie or browser related problem.
As you can see it just disappears after reloading the website.

https://user-images.githubusercontent.com/46091052/204400888-caef5086-963c-4c24-a00b-ba408fb30569.mp4

**Comment by ahmedalashhab** [comment]
Ill look into this soon.

**Comment by K3NOXOFFICIAL** [comment]
@ahmedalashhab Was something changed in firebase on Ashanime?

On Ashanime it now suddently seems to work but on my Own deployment and all Forks it still doesn't.
If you changed somthing can you send detailed what you changed or can you send your firebase configurations so I can look into whats configured wrong.

If you have time to look into this issue look into the mail I sent you + I invited you into my Fork of Ashanime.
Maybe we can resolve the issue like that.

**Comment by ahmedalashhab** [comment]
@K3NOXOFFICIAL Hey there. The personal data syncing feature has always been working with my app on my site. Perhaps you are doing something wrong? I haven't changed anything on my end. I'll have a look into your repo.

Can you please send me a screenshot of your firebase user accounts data?
<img width="745" alt="image" src="https://user-images.githubusercontent.com/96653465/205129192-bd900601-95c7-40c1-bbbe-d2a74ced49dc.png">
It's like the one above. 
I need to know if any data is actually being sent to your firebase database.

**Comment by ahmedalashhab** [comment]
So I tested your firebase account and the issue seems to do with this particular stackoverflow problem: https://stackoverflow.com/questions/68680998/firebase-warning-missing-appcheck-token-https-xxxxxxxx-firebaseio-com

TLDR; It seems you may need to disable Appcheck on your firebase project, https://firebase.google.com/docs/app-check

The above may guide you to your issue, or, it could be the below.

Access your firebase Authentication, then settings, then Authorized Domains, and add your personal domain which is https://www.k3noxofficial.net/ in case you haven't.

**Comment by K3NOXOFFICIAL** [comment]
@ahmedalashhab i turned it on because i was unsure if the issue is comming from having it turned off.
I turned it on only today so the issue must come from antother reason / problem.
Ill turn it off immidietly.
Thanks really much for the help tho.

**Comment by K3NOXOFFICIAL** [comment]
![image](https://user-images.githubusercontent.com/46091052/205165550-d677edb0-c2ff-4b79-a8d3-cea6cd48e8ac.png)

![image](https://user-images.githubusercontent.com/46091052/205166210-dd4cf447-4f42-408a-b239-e95c30b4d6a8.png)

![image](https://user-images.githubusercontent.com/46091052/205167341-c296463a-9a0f-41b5-8f3a-df751477027a.png)


@ahmedalashhab i turned it on because i was unsure if the issue is comming from having it turned off.
I turned it on only today so the issue must come from antother reason / problem.
Ill turn it off immidietly.
Maybe it got something to do with these 2 things?

Thanks really much for the help tho.

**Comment by K3NOXOFFICIAL** [comment]
@ahmedalashhab Short Update: It was indeed the Realtime Database Rules.
I set read and write to true and it now works if there is additional set up to do in rules please update it to your Readme so nobody else has the same problem.

Thanks a lot

**Comment by ahmedalashhab** [comment]
That is a great idea. Use these rules 

{
  "rules": {
    "users": {
      "$userId": {
        // grants write access to the owner of this user account
        // whose uid must exactly match the key ($userId)
        ".write": "$userId === auth.uid",
        ".read" : "$userId === auth.uid"
      }
    }
  }
}

**Comment by K3NOXOFFICIAL** [comment]
@ahmedalashhab Thanks a lot I'll Close the issue cause you already updated the Readme-
